src/db/schema.ts

```diff
@@ -57,72 +69,117 @@ export const messagesTable = sqliteTable("messages", {
+    ...timestamps
+  },
+  (table) => ({
+    pk: primaryKey({ columns: [table.user_id, table.contact_id] })
+  })
+)
 
-export type InsertUserChat = typeof userChatsTable.$inferInsert;
-export type SelectUserChat = typeof userChatsTable.$inferSelect;
+export const useContactsRelations = relations(userContactsTable, ({ one }) => ({
+  user: one(usersTable, {
+    relationName: "userToContacts",
+    fields: [userContactsTable.user_id],
+    references: [usersTable.unique_id]
+  }),
+  contact: one(usersTable, {
+    relationName: "contactToUser",
+    fields: [userContactsTable.contact_id],
+    references: [usersTable.unique_id]
+  })
+}))
+
+export const tagsTable = sqliteTable("tags", {
+  id: int().primaryKey({ autoIncrement: true }),
+  name: text().notNull(),
+  type: text().notNull(),
+  ...timestamps
+})
 
-export const userMessagesTable = sqliteTable("user_messages", {
+export const userTagsTable = sqliteTable("user_tags", {
+  id: int().primaryKey({ autoIncrement: true }),
   user_id: text().notNull(),
-  message_id: int().notNull(),
-});
+  tag_id: int().notNull()
+})
 
-export type InsertUserMessage = typeof userMessagesTable.$inferInsert;
-export type SelectUserMessage = typeof userMessagesTable.$inferSelect;
+export const rewardsTable = sqliteTable("rewards", {
+  id: int().primaryKey({ autoIncrement: true }),
+  title: text().notNull(),
+  description: text().notNull(),
+  badge_type: text().notNull(),
+  ...timestamps
+})
 
-export const userContactsTable = sqliteTable("user_contacts", {
+export const userRewardsTable = sqliteTable("user_rewards", {
+  id: int().primaryKey({ autoIncrement: true }),
   user_id: text().notNull(),
-  contact_id: text().notNull(),
-  is_requested: int().notNull().default(0),
-  is_accepted: int().notNull().default(0),
-  is_blocked: int().notNull().default(0),
-  is_following: int().notNull().default(0),
-  ...timestamps,
-  
-}, (t) => ({
-  pk: primaryKey({columns: [t.user_id, t.contact_id] }),
-}));
-
-export const useContactsRelations = relations(userContactsTable, ({ one }) => ({
-	user: one(usersTable, {
-		fields: [userContactsTable.user_id],
-		references: [usersTable.unique_id],
-	}),
-	contact: one(usersTable, {
-		fields: [userContactsTable.contact_id],
-		references: [usersTable.unique_id],
-	}),
-}));
-
-export type InsertUserContact = typeof userContactsTable.$inferInsert;
-export type SelectUserContact = typeof userContactsTable.$inferSelect;
+  reward_id: int().notNull()
+})
 
+export const activitiesTable = sqliteTable("activities", {
+  id: int().primaryKey({ autoIncrement: true }),
+  title: text().notNull(),
+  date: text().notNull(),
+  description: text().notNull(),
+  type: text().notNull(),
+  ...timestamps
+})
 
+export const userActivitiesTable = sqliteTable("user_activities", {
+  id: int().primaryKey({ autoIncrement: true }),
+  user_id: text().notNull(),
+  activity_id: int().notNull()
+})
+
+export type InsertUser = typeof usersTable.$inferInsert
```

Comment:
dont make all the types in acculating all in one place 
very difficult to see the whole image